WalletWasabi.Fluent/ViewModels/ApplicationViewModel.cs

```diff
@@ -13,33 +15,41 @@
 
 namespace WalletWasabi.Fluent.ViewModels;
 
-public class ApplicationViewModel : ViewModelBase, ICanShutdownProvider
+public partial class ApplicationViewModel : ViewModelBase, ICanShutdownProvider
 {
-	public ApplicationViewModel()
+	private readonly IMainWindowService _mainWindowService;
+	[AutoNotify] private bool _isMainWindowShown = true;
+
+	public ApplicationViewModel(IMainWindowService mainWindowService)
 	{
-		QuitCommand = ReactiveCommand.CreateFromTask(async () =>
-		{
-			if (CanShutdown())
+		_mainWindowService = mainWindowService;
+
+		this.WhenAnyValue(x => x.IsMainWindowShown)
+			.Where(x => x == false)
+			.Subscribe(async x =>
 			{
-				if (Application.Current?.ApplicationLifetime is IControlledApplicationLifetime lifetime)
+				if (!Services.UiConfig.HideOnClose)
 				{
-					lifetime.Shutdown();
+					DoQuit();
 				}
+			});
```

Comment:
GitHub doesn't allow me to suggest this:
```
this.WhenAnyValue(x => x.IsMainWindowShown)
	.Where(x => x == false && !Services.UiConfig.HideOnClose)
	.Subscribe(_ => DoQuit());
```

- `async x` not needed.
- move the condition above to reduce nesting